### PR TITLE
fix(1320): fix sidebar tarball protocol

### PR DESCRIPTION
**Pull request by honzahommer**
**Type:** Fix

**Resolves:** https://github.com/verdaccio/verdaccio/issues/1320

### CODE REVIEW

File: src/api/web/endpoint/package.js [hunk 1]
```diff
@@ -102,7 +101,8 @@ function addPackageWebApi(route: Router, storage: IStorageHandler, auth: IAuth,
       callback: function(err: Error, info: $SidebarPackage) {
         if (_.isNil(err)) {
           let sideBarInfo: any = _.clone(info);
-          sideBarInfo.latest = info.versions[info[DIST_TAGS].latest];
+          sideBarInfo.versions = convertDistRemoteToLocalTarballUrls(info, req, config.url_prefix).versions;
```
Comment (juanpicado) [inline src/api/web/endpoint/package.js]: Remember the `config.url_prefix` does not work with URL anymore, check https://github.com/verdaccio/verdaccio/pull/1299

Comment (honzahommer) [reply src/api/web/endpoint/package.js]: > Remember the `config.url_prefix` does not work with URL anymore with `4.0.0`, check #1299

I know, but I've made copy-paste from... https://github.com/verdaccio/verdaccio/blob/f0b8673ecfdc23b7f73a86a40b8a25f63367d32f/src/api/endpoint/api/package.js#L36

**Comment by honzahommer** [comment]
> I think the issue on #1320 is a different thing. Either the tarball was uploaded without HTTPS and then the server was moved to HTTPS and metadata now does not match.

This should be an issue... https://github.com/verdaccio/verdaccio/blob/9bc95974137de860739dfc68952389d10ec221dc/src/lib/local-storage.js#L759-L778

**Comment by juanpicado** [comment]
> _updateUplinkToRemoteProtocol

Where exactly? @honzahommer

**Comment by juanpicado** [comment]
> > I think the issue on #1320 is a different thing. Either the tarball was uploaded without HTTPS and then the server was moved to HTTPS and metadata now does not match.
> 
> @juanpicado This should be an issue...
> 
> https://github.com/verdaccio/verdaccio/blob/9bc95974137de860739dfc68952389d10ec221dc/src/lib/local-storage.js#L759-L778

@honzahommer interesting, even if we set HTTPS in our registry, but having on `.npmrc`

```
ca=null
https_proxy=https://example.com
strict-ssl=false
```

```
 npm publish --registry https://example.com --verbose 
npm info it worked if it ends with ok
npm verb cli [ '/Users/xxx/.nvm/versions/node/v10.15.0/bin/node',
npm verb cli   '/Users/xxx/.nvm/versions/node/v10.15.0/bin/npm',
npm verb cli   'publish',
npm verb cli   '--registry',
npm verb cli   'https://example.com',
npm verb cli   '--verbose' ]
npm info using npm@6.9.0
npm info using node@v10.15.0
npm verb npm-session 364f1b677abd531d
npm verb publish [ '.' ]
npm info lifecycle juan_test_webpack9@10.0.0~prepublish: juan_test_webpack9@10.0.0
npm info lifecycle juan_test_webpack9@10.0.0~prepare: juan_test_webpack9@10.0.0
npm info lifecycle juan_test_webpack9@10.0.0~prepublishOnly: juan_test_webpack9@10.0.0
npm info lifecycle juan_test_webpack9@10.0.0~prepack: juan_test_webpack9@10.0.0
npm info lifecycle juan_test_webpack9@10.0.0~postpack: juan_test_webpack9@10.0.0
npm notice 
npm notice 📦  juan_test_webpack9@10.0.0
npm notice === Tarball Contents === 
npm notice 221B  package.json  
npm notice 6.1kB .DS_Store     
npm notice 0     index.js      
npm notice 6B    README.md     
npm notice 1.3kB yarn-error.log
npm notice === Tarball Details === 
npm notice name:          juan_test_webpack9                      
npm notice version:       10.0.0                                  
npm notice package size:  1.2 kB                                  
npm notice unpacked size: 7.7 kB                                  
npm notice shasum:        10afb700e60e272c259745a5f937048e058985c0
npm notice integrity:     sha512-6QN7l1CbKzJDz[...]IVs/+S4Sr0oyA==
npm notice total files:   5                                       
npm notice 
```

The metadata sent my npm cli sends the tarball url as HTTP instead HTTPS, thus, we save the tarball as HTTP, which, your patch fix that on sidebar, which is correct. 

`http://example.com/juan_test_webpack9/-/juan_test_webpack9-10.0.0.tgz`

Screen Shot 2019-05-28 at 10 10 52 PM

**Pull request merged**